applications/serial_lte_modem/src/lwm2m_carrier/slm_at_carrier.c

```diff
@@ -861,6 +953,647 @@ int handle_at_carrier(enum at_cmd_type cmd_type)
+
+	count = at_params_valid_count_get(&slm_at_param_list);
+	if (count == 2) {
+		rsp_send("\r\n#XCARRIERCFG: %s\r\n", lwm2m_settings_software_version_get());
+		return 0;
+	}
+
+	int ret;
+	char software_version[32];
+	int size = sizeof(software_version);
+
+	ret = util_string_get(&slm_at_param_list, 2, software_version, &size);
+	if (ret) {
+		return ret;
+	}
+
+	return lwm2m_settings_software_version_set(software_version);
+}
+
+/* AT#XCARRIERCFG="session_idle_timeout"[,<timeout>] */
+static int do_cfg_session_idle_timeout(void)
+{
+	uint32_t count;
+
+	count = at_params_valid_count_get(&slm_at_param_list);
+	if (count == 2) {
+		rsp_send("\r\n#XCARRIERCFG: %d\r\n", lwm2m_settings_session_idle_timeout_get());
+		return 0;
+	}
+
+	int ret, session_idle_timeout;
+
+	ret = at_params_int_get(&slm_at_param_list, 2, &session_idle_timeout);
+	if (ret) {
+		return ret;
+	}
+
+	if (session_idle_timeout < -1 || session_idle_timeout > 86400) {
+		LOG_DBG("AT#XCARRIERCFG=\"session_idle_timeout\" failed: value must be "
+			"-1 or between 0 and 86400 (24 hours)");
+		return -EINVAL;
+	}
+
+	return lwm2m_settings_session_idle_timeout_set(session_idle_timeout);
+}
+
+/* AT#XCARRIERCFG="device_serial_no_type"[,<0|1>] */
+static int do_cfg_device_serial_no_type(void)
+{
+	uint32_t count;
+
+	count = at_params_valid_count_get(&slm_at_param_list);
+	if (count == 2) {
+		rsp_send("\r\n#XCARRIERCFG: %u\r\n", lwm2m_settings_device_serial_no_type_get());
+		return 0;
+	}
+
+	int ret, serial_no_type;
+
+	ret = at_params_int_get(&slm_at_param_list, 2, &serial_no_type);
+	if (ret) {
+		return ret;
+	}
+
+	switch (serial_no_type) {
+	case 0:
+		serial_no_type = LWM2M_CARRIER_LG_UPLUS_DEVICE_SERIAL_NO_IMEI;
+		break;
+	case 1:
+		serial_no_type = LWM2M_CARRIER_LG_UPLUS_DEVICE_SERIAL_NO_2DID;
+		break;
+	default:
+		LOG_DBG("AT#XCARRIERCFG=\"device_serial_no_type\" failed: value must be 0 or 1");
+		return -EINVAL;
+	}
+
+	return lwm2m_settings_device_serial_no_type_set(serial_no_type);
+}
+
+/* AT#XCARRIERCFG="service_code"[,<service_code>] */
+static int do_cfg_service_code(void)
+{
+	uint32_t count;
+
+	count = at_params_valid_count_get(&slm_at_param_list);
+	if (count == 2) {
+		rsp_send("\r\n#XCARRIERCFG: %s\r\n", lwm2m_settings_service_code_get());
+		return 0;
+	}
+
+	int ret;
+	char service_code[5];
+	int size = sizeof(service_code);
+
+	ret = util_string_get(&slm_at_param_list, 2, service_code, &size);
+	if (ret) {
+		return ret;
+	}
+
+	return lwm2m_settings_service_code_set(service_code);
+}
+
+/* AT#XCARRIERCFG="pdn_type"[,<pdn_type>] */
+static int do_cfg_pdn_type(void)
+{
+	uint32_t count;
+
+	count = at_params_valid_count_get(&slm_at_param_list);
+	if (count == 2) {
+		rsp_send("\r\n#XCARRIERCFG: %d\r\n", lwm2m_settings_pdn_type_get());
+		return 0;
+	}
+
+	int ret;
+	uint16_t pdn_type;
+
+	ret = at_params_unsigned_short_get(&slm_at_param_list, 2, &pdn_type);
+	if (ret) {
+		return ret;
+	}
+
+	if (pdn_type > LWM2M_CARRIER_PDN_TYPE_NONIP) {
```

Comment:
Document these as well?

Reply:
To avoid duplicating documentation, I have now referred to the respective sections of the library's doc, let me know if that's ok. Maybe @wiba-nordic has some suggestions too.

Reply:
Referring to the documentation is probably the best strategy. Linking directly to the documentation would be the best, but that doesn't work well with versioning since we'd have to update all links for each release to be release-specific.

Reply:
I have added documentation for any AT command inputs that are not used directly in the function calls but mapped into appropriate values first.